File: clients/dotnet/models/ErrorDetail.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;
using Newtonsoft.Json;

namespace Avalara.AvaTax.RestClient
{
    /// <summary>
    /// Message object
    /// </summary>
    public class ErrorDetail
    {
        /// <summary>
        /// Name of the error.
        /// </summary>
        public ErrorCodeId? code { get; set; }

        /// <summary>
        /// Error message identifier
        /// </summary>
        public Int32? number { get; set; }

        /// <summary>
        /// Concise summary of the message, suitable for display in the caption of an alert box.
        /// </summary>
        public String message { get; set; }

        /// <summary>
        /// A more detailed description of the problem referenced by this error message, suitable for display in the contents area of an alert box.
        /// </summary>
        public String description { get; set; }

        /// <summary>
        /// Indicates the SoapFault code
        /// </summary>
        public String faultCode { get; set; }

        /// <summary>
        /// URL to help for this message
        /// </summary>
        public String helpLink { get; set; }

        /// <summary>
        /// Item the message refers to, if applicable.  This is used to indicate a missing or incorrect value.
        /// </summary>
        public String refersTo { get; set; }

        /// <summary>
        /// Severity of the message
        /// </summary>
        public SeverityLevel? severity { get; set; }


    }
}

File: clients/php/models/ErrorDetail.php
<?php 
namespace Avalara;
/*
 * AvaTax API Client Library
 *
 * (c) 2004-2016 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @category   AvaTax client libraries
 * @package    Avalara.AvaTaxClient
 * @author     Ted Spence <ted.spence@avalara.com>
 * @author     Bob Maidens <bob.maidens@avalara.com>
 * @copyright  2004-2016 Avalara, Inc.
 * @license    https://www.apache.org/licenses/LICENSE-2.0
 * @version    2.16.12-30
 * @link       https://github.com/avadev/AvaTaxClientLibrary
 */


/**
 * Message object
 */
final class ErrorDetail
{
    /**
     * @var ErrorCodeId? Name of the error.
     */
    public $code;

    /**
     * @var Int32? Error message identifier
     */
    public $number;

    /**
     * @var String Concise summary of the message, suitable for display in the caption of an alert box.
     */
    public $message;

    /**
     * @var String A more detailed description of the problem referenced by this error message, suitable for display in the contents area of an alert box.
     */
    public $description;

    /**
     * @var String Indicates the SoapFault code
     */
    public $faultCode;

    /**
     * @var String URL to help for this message
     */
    public $helpLink;

    /**
     * @var String Item the message refers to, if applicable.  This is used to indicate a missing or incorrect value.
     */
    public $refersTo;

    /**
     * @var SeverityLevel? Severity of the message
     */
    public $severity;

}
